### PR TITLE
legend indentation

### PR DESCRIPTION
**Pull request by my-tien**
# legend indentation

This PR adds the property `indentation` to `layout.legend` so that legend entries can be indented relative to the title.

I have set the min value of this to 0 for now to avoid having to deal with adjusting the legend width and title positioning when the legend extends towards the left.

### CODE REVIEW

File: src/components/legend/attributes.js [hunk 1]
```diff
@@ -105,6 +105,13 @@ module.exports = {
         editType: 'legend',
         description: 'Determines what entrywidth means.',
     },
+    indentation: {
+        valType: 'number',
+        min: 0,
```
Comment (alexcjohnson) [inline src/components/legend/attributes.js:110]: I could see us allowing negative indentation, to remove or reduce the padding to the left of many of the symbols. Many if not most legend symbols don't use the full `itemwidth`. The risk here is allowing symbols to bleed off the edge of the legend if you go too far, but I think that's acceptable if you have explicitly set a negative indentation. Maybe `min: -15`, ie half the minimum `itemwidth`?

Comment (my-tien) [reply src/components/legend/attributes.js]: Yes the bleeding off the left edge was my concern as well. I have adapted the test image to showcase how it looks to indent with -15 for a line and a bar legend entry.

Comment (alexcjohnson) [reply src/components/legend/attributes.js]: Oh cool, we clip at the edge of the legend, that's perfect.

**Review by alexcjohnson — APPROVED** [review]
💃 Great job @my-tien!

**Pull request merged**